aleph-client/src/lib.rs

```diff
@@ -36,23 +36,26 @@ pub type Header = GenericHeader<BlockNumber, BlakeTwo256>;
 pub type KeyPair = sr25519::Pair;
 pub type Connection = Api<KeyPair, WsRpcClient>;
 
-pub fn create_connection(address: &str) -> Connection {
-    create_custom_connection(address).expect("connection should be created")
+pub fn create_connection(address: &str, ssl: bool) -> Connection {
```

Comment:
We can consider not using bool arguments in public API functions, this is because it's A) easy to make swap error as a caller when they are more than one bool params (not this case here but making a general point) and B) without peeking into docs harder to guess what those bool args are about). E.g. consider `pub fn hedge_assets(native_currency: bool, buy: bool` as caller `hedge_assets(true, false) -without peeking into docs. The alternative is to expose public API functions with proper naming e.g.
```rust
pub fn create_connection_secured(address: &str) -> Connection {
     create_connection(address, true)
}

pub fn create_connection_not_secured(address: &str) -> Connection {
     create_connection(address, false)
}

fn create_connection(address: &str, ssl: bool) -> Connection {
// ....
}
```

Reply:
yes, I was thinking about it, but gave up this because then, there is a need of putting this ugly conditional in many places:
```
if config.ssl { create_connection_secured } else { create_connection_not_secured }
```
and this imho is worse, because we have this semantics in methods, that should not care much; for now we are just passing this argument from the user to the end, so it is not so perilous

anyway, maybe using `enum` as @kostekIV suggested is a compromise?

Reply:
Afaik in `e2e-client` we can leave by default `create_connection_not_secured` in all code, so does in `benches/payout_stakers`.  In `cliain` we'd need to pass it but just once, see the other comment to pass `Connection&` to `cliain` command methods.

Reply:
And yes , either enums or named functions are good for me.

Reply:
imho with clear naming the problem dissappears, e.g `use_ssl: true/false`

Reply:
I changed to `enum`. The only inconvenience is that I need to implement some kind of parsing for each cli tool (I cannot derive `ArgEnum` on this `enum` as it would require adding `clap` dependency to `aleph-client`)